src/libs/Network/NetworkStore.js

```diff
@@ -80,7 +80,7 @@ Onyx.connect({
             triggerReconnectCallback();
         }
 
-        offline = network.isOffline;
+        offline = Boolean(network.shouldForceOffline) || network.isOffline;
```

Comment:
In my opinion, the switch `network.shouldForceOffline` should emulate going offline/online as close as possible to the real thing. With the current code, switching `network.shouldForceOffline` to `true` and then `false` should call `triggerReconnectCallback`, but it is not at the moment.

I think the code:

```js
        if (offline && !network.isOffline) {
            triggerReconnectCallback();
        }
```
should be something like

```js
        if ((offline || shouldForceOffline) && (!network.shouldForceOffline && ! network.isOffline) {
            triggerReconnectCallback();
        }
```

Reply:
Yes I agree that it should trigger the reconnection callbacks. Is that not the case? I added a log line and I can see that it's called.
https://user-images.githubusercontent.com/26260477/204056631-9b516080-70ba-4630-a7eb-663cc291b68b.mov



When we enable `shouldForceOffline` the `offline` variable will be set to `true`. When `shouldForceOffline` is disabled again `offline` will be `true` and `!network.isOffline` will also be true so we will trigger the reconnect callback.

The reason I included `shouldForceOffline` here is that when the App starts up `network.isOffline` is initialized to `false`.
https://github.com/Expensify/App/blob/1bbe8aea89daf7a8805b9bfd6ec753ca0278f7ae/src/libs/Network/NetworkStore.js#L83
https://github.com/Expensify/App/blob/360fa2033bfc7f8eb41c4c87a3603b35aa6152b1/src/setup/index.js#L34

Reply:
> Yes I agree that it should trigger the reconnection callbacks. Is that not the case? I added a log line and I can see that it's called.

Oh, I'll check again. I was watching the Network requests when I tested.. but maybe I did something wrong!

Reply:
are you looking at the `OpenReport` request? the one I'm talking about is `ReconnectApp`


https://user-images.githubusercontent.com/87341702/204068159-5fe9d47b-eb75-460a-8247-07ea7e208d7a.mov

Reply:
It is really strange because it is being called for me. 

https://user-images.githubusercontent.com/83179295/204123636-2678e1ec-984c-461d-a475-4af2865df85b.mov

Reply:
Ah got it, so you're concerned about ReconnectApp. I'll take a look into why that's not running.

Reply:
> It is really strange because it is being called for me.

That is `ReconnectToReport` which I didn't know existed.. it may be new. I'm talking about `ReconnectApp`

Reply:
Thanks for catching that! I wasn't using NetworkConnection.setOfflineStatus, so the reconnection callback weren't triggered. It works properly now. I'll add this to the testing steps.